### ISSUE TITLE
Error codes management

### ISSUE THREAD

**Issue by prb28** [emoon/fs-uae#3]
The only returned error code is E1.
It would be interesting to have a managed list of error codes.

**Comment by emoon** [comment]
For sure. This is part of the "the code isn't finished at all" section :)

**Comment by emoon** [comment]
So I read up a bit on this:

**Comment by emoon** [comment]
So I read up a bit on this

```
Error response
(“E”)
When a debugging stub encounters an
error  during  command  processing,  it
should send an error response back to
gdb. Bus errors and/or illegal address-
es  during  memory  operations  are
examples  of  commands  which  may
generate  an  error  response  from  a
debugging stub.
Example: 
<a command that produces
an error>
Target reponse: 
+$E01#xx
There  aren’t  any  predefined  error
codes  in  gdb;  when  gdb  receives  an
error  message,  it  prints  it  on  the  con-
sole   and   halts   the   operation   in
progress

**Comment by emoon** [comment]
Given this I guess we can just define our own error codes.

**Comment by prb28** [comment]
I use this source for the protocol: https://sourceware.org/gdb/onlinedocs/gdb/Remote-Protocol.html#Remote-Protocol
And I agree with you it is an internal code, and it may be chosen by the implementation.

Last time I did it in C (years ago...), I used a list of #define in a header file, is that the way you would do it?

The code works pretty well and I don't think it's too far from being ready for a first basic release.
It is your project, and should be done your way, under your lead.
I can give a hand and add some features, but it's a hobby so it will be slow.
Do you have a TODO list in mind ? 
(I will add some in the issues of this project - I'm using my plugin with the photon/scoopex tutorial to see what is missing)

**Comment by emoon** [comment]
Having a header is likely good as it make it a bit easier for someone looking over the error codes to have them in one place.

I really don't have a clear TODO list in my mind currently. I can try to add some issues during the day that I know I want to get fixed.

**Comment by prb28** [comment]
That would be great I'll see if I can fix some of them.

**Comment by emoon** [comment]
I have added some now. I think this one is the most important currently https://github.com/emoon/fs-uae/issues/12 That is if there is a crash we should be able to figure out where it happens.

**Comment by prb28** [comment]
I agree, the #12 seems to be the most important.

**Comment by emoon** [comment]
Yeah so I think #12 needs to be supported for the first release. The two main ways (imo) to use a debugger (for Amiga/Native) is to step code and inspect a crash. It's fairly simple to crash Amiga as there is no memory protection so it would be nice to catch crashes when it happens in the debugger.
